<?xml version="1.0"?>
<launch>
  <arg name="device" default="0"/>
  <arg name="width" default="160"/>
  <arg name="height" default="120"/>
  <arg name="frame_rate" default="8"/>

  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node"
      output="screen"
      if="true">
    <param name="image_width" value="$(arg width)"/>
    <param name="image_height" value="$(arg height)"/>
    <param name="framerate" value="$(arg frame_rate)"/>
    <param name="video_device" value="/dev/video$(arg device)" />
    <!--param name="pixel_format" value="mjpeg" /-->
    <param name="pixel_format" value="yuyv"/>
  </node>
  <node name="v4l2ucp" pkg="v4l2ucp" type="v4l2ucp" >
    <param name="device" value="/dev/video$(arg device)" />
  </node>
  <!-- TODO(lucasw) Need to delay launch of this until the v4l2ucp node above
    has complete creating all the parameters- maybe it should publish an 'update'
    signal? -->
  <node name="dr_topics" pkg="dynamic_reconfigure_tools" type="dr_topics.py" >
  </node>


  <node pkg="image_publisher" type="image_publisher"
    name="load0"
    args="$(find vimjay)/data/clouds.png"
    if="false">
    <param name="publish_rate" value="0.5"/>
  </node>

  <group ns="usb_cam">
    <group ns="image_raw">
    <!-- http://answers.ros.org/question/255970/how-to-deactivate-image_transport-plugins/ -->
    <rosparam param="disable_pub_plugins">
      - 'image_transport/compressed'
      - 'image_transport/compressedDepth'
      - 'image_transport/theora'
    </rosparam>
    </group>
    <node pkg="image_manip" type="resize_node"
        name="resize0"
        output="screen">
      <param name="frame_rate" value="$(arg frame_rate)"/>
      <param name="width" value="$(arg width)"/>
      <param name="height" value="$(arg height)"/>
      <remap from="image_in" to="image_raw"/>
      <remap from="image_out" to="resized"/>
    </node>
  </group>

</launch>
